transcription/transcription.go

```diff
@@ -38,6 +45,49 @@ func msgHeaders(from string, to []string, subject string) string {
 	return strings.Join(msgHeaders, "\r\n")
 }
 
+// StartTranscription transcribes a given file using Sphinx.
+// File name should be in "name.wav" format.
+func StartTranscription(fileName string, command string) (string, error) {
+
+	cmd := exec.Command("java", "-jar", command, fileName)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+	outputFileName := "/Sphinx/files/" + fileName + "-json.txt"
+
+	// //once json is sent somewhere, capture the output
+	// if _, err := transcriptionOutputToStruct(outputFileName); err != nil {
+	// 	return err
+	// }
+	return outputFileName, nil
+}
+
+// transcriptionOutputToStruct takes a text file and reads its input
+// into a Go struct
+func transcriptionOutputToStruct(fileName string) (transcription, error) {
+	var jsonData transcription
+	file, err := os.Open(fileName)
+	r := bufio.NewReader(file)
+
+	bytesText, err := r.ReadBytes('\n')
+	if err != nil {
+		return jsonData, err
+	}
+	bytesMeta, err := r.ReadBytes('\n')
+	if err != nil || err != io.EOF {
+		return jsonData, err
+	}
+
+	nText := bytes.IndexByte(bytesText, 0)
+	nMeta := bytes.IndexByte(bytesMeta, 0)
+	sText := string(bytesText[:nText])
+	sMeta := string(bytesText[:nMeta])
```

Comment:
Which built in json package? I am reading in the data from a text file, which is how Sphinx stores it. I'm reading each line and storing the data in the appropriate struct fields .